src/main/java/org/springframework/session/data/mongo/ReactiveMongoOperationsSessionRepository.java

```diff
@@ -107,9 +107,18 @@ public ReactiveMongoOperationsSessionRepository(ReactiveMongoOperations mongoOpe
 	@Override
 	public Mono<Void> save(MongoSession session) {
 
-		return this.mongoOperations
-			.save(convertToDBObject(this.mongoSessionConverter, session), this.collectionName)
-			.then();
+		if (session.isNew()) {
+
+			session.setNew(false);
+			return this.mongoOperations.save(convertToDBObject(this.mongoSessionConverter, session), this.collectionName)
+				.then();
+		} else {
+
+			return findSession(session.getId())
```

Comment:
While this works, you could do probably do a `ReactiveMongoOperations#collectionExists` to optimize by removing deserialization cost from the mix, right?

You could also compose the implementation of this method in such way to avoid writing the duplicate `ReactiveMongoOperations#save` call - see [`ReactiveRedisOperationsSessionRepository#save`](https://github.com/spring-projects/spring-session/blob/0e89539e203ccd5dd7ed97832211a0f7a0a6d918/spring-session-data-redis/src/main/java/org/springframework/session/data/redis/ReactiveRedisOperationsSessionRepository.java#L145) for inspiration.